File: connectors-editors/redmine-editor/src/test/java/com/taskadapter/connector/redmine/editor/RedmineEditorFactoryTest.java
package com.taskadapter.connector.redmine.editor;

import com.taskadapter.connector.definition.WebConnectorSetup;
import com.taskadapter.connector.definition.exceptions.ProjectNotSetException;
import com.taskadapter.connector.redmine.RedmineConfig;
import com.taskadapter.connector.redmine.RedmineConnector;
import com.taskadapter.editor.testlib.VaadinTestHelper;
import com.taskadapter.web.service.Sandbox;
import com.vaadin.flow.data.binder.Binder;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.TemporaryFolder;

import java.util.Collections;

import static org.assertj.core.api.Assertions.assertThat;

public class RedmineEditorFactoryTest {

    @Rule
    public TemporaryFolder tempFolder = new TemporaryFolder();

    @Before
    public void beforeEachTest() {
        VaadinTestHelper.initVaadinSession(RedmineEditorFactoryTest.class);
    }

    private RedmineEditorFactory factory = new RedmineEditorFactory();
    private RedmineConfig config = new RedmineConfig();
    private WebConnectorSetup setup = WebConnectorSetup.apply(RedmineConnector.ID, "label1", "http://somehost", "", "", false, "");
    private Binder<RedmineConfig> binder = new Binder<>();

    @Test
    public void miniPanelIsCreated() {
        factory.getMiniPanelContents(new Sandbox(true, tempFolder.getRoot()), new RedmineConfig(), setup);
    }

    @Test
    public void givesErrorForEmptyProjectKey() {
        var exceptions = factory.validateForSave(new RedmineConfig(), setup, Collections.emptyList());
        assertThat(exceptions.get(0)).isInstanceOf(ProjectNotSetException.class);
    }

    @Test
    public void passesWithSomeProjectKey() {
        config.setProjectKey("project123");
        var errors = factory.validateForSave(config, setup, Collections.emptyList());
        assertThat(errors).isEmpty();
    }
}

File: connectors-editors/redmine-editor/src/test/scala/com/taskadapter/connector/redmine/editor/RedmineEditorFactoryTest.scala
package com.taskadapter.connector.redmine.editor

import com.taskadapter.connector.definition.WebConnectorSetup
import com.taskadapter.connector.definition.exceptions.ProjectNotSetException
import com.taskadapter.connector.redmine.{RedmineConfig, RedmineConnector}
import com.taskadapter.connector.testlib.TempFolder
import com.taskadapter.editor.testlib.VaadinTestHelper
import com.taskadapter.web.service.Sandbox
import com.vaadin.flow.data.binder.Binder
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}


@RunWith(classOf[JUnitRunner])
class RedmineEditorFactoryTest extends FunSpec with Matchers with TempFolder {

  VaadinTestHelper.initVaadinSession(getClass)
  val factory = new RedmineEditorFactory
  val config = new RedmineConfig()
  val setup = WebConnectorSetup(RedmineConnector.ID, "label1", "http://somehost", "", "", false, "")
  val binder = new Binder[RedmineConfig]

  it("mini panel is created") {
    withTempFolder { folder =>
      factory.getMiniPanelContents(new Sandbox(true, folder), new RedmineConfig, setup)
    }
  }
  describe("config validation for save") {
    it("gives error for empty project key") {
      val exceptions = factory.validateForSave(new RedmineConfig, setup, Seq())
      exceptions.head shouldBe a[ProjectNotSetException]
    }
    it("passes with some project key") {
      config.setProjectKey("project123")
      val errors = factory.validateForSave(config, setup, Seq())
      errors shouldBe empty
    }
  }
}